Stratis.Bitcoin/Configuration/Logging/LoggingConfiguration.cs

```diff
@@ -200,7 +171,97 @@ public static void AddConsoleWithFilters(this ILoggerFactory loggerFactory)
                 }
             };
 
-            loggerFactory.AddConsole(settings);
+            loggerFactory.AddConsole(consoleLoggerSettings);
+        }
+
+        /// <summary>
+        /// Configure the console logger and set it to filter logs not related to the fullnode.
+        /// </summary>
+        /// <param name="loggerFactory">Not used.</param>
+        /// <param name="consoleLoggerSettings">Console settings to filter.</param>
+        /// <param name="settings">A settings that hold potential debug arguments, if null no debug arguments will be loaded."/></param>
+        public static void ConfigureConsoleFilters(this ILoggerFactory loggerFactory, ConsoleLoggerSettings consoleLoggerSettings, LogSettings settings)
+        {
+            if (settings != null)
+            {
+                if (settings.DebugArgs.Any())
+                {
+                    if (settings.DebugArgs[0] == "1")
+                    {
+                        // Increase all logging to Trace.
+                        consoleLoggerSettings.Switches.Add($"{nameof(Stratis)}.{nameof(Stratis.Bitcoin)}", Microsoft.Extensions.Logging.LogLevel.Trace);
+                    }
+                    else
+                    {
+                        var keyToCategory = LoadKeyCategorys();
+                        HashSet<string> usedCategories = new HashSet<string>(StringComparer.Ordinal);
+
+                        // Increase selected categories to Trace.
+                        foreach (string key in settings.DebugArgs)
+                        {
+                            string category;
+                            if (!keyToCategory.TryGetValue(key.Trim(), out category))
+                            {
+                                // Allow direct specification - e.g. "-debug=Stratis.Bitcoin.Miner".
+                                category = key.Trim();
+                            }
+
+                            if (!usedCategories.Contains(category))
+                            {
+                                usedCategories.Add(category);
+                                consoleLoggerSettings.Switches.Add(category.TrimEnd('*').TrimEnd('.'), Microsoft.Extensions.Logging.LogLevel.Trace);
+                            }
+                        }
+                    }
+                }
+            }
+
+            consoleLoggerSettings.Reload();
+        }
+
+        /// <summary>
+        /// Create a key to category mappings to allow to limit filtering based on short debug codes
+        /// </summary>
+        /// <returns>The key category mappings.</returns>
+        private static Dictionary<string, string> LoadKeyCategorys()
```

Comment:
Would it not be better to have this as a static field with initializer on the class instead of method?